### PR TITLE
Merged consul and config-seed into one docker 

### PR DESCRIPTION
**Pull request by chenchix**
To avoid high memory peaks during docker deployment, we have merged consul configuration and config-seed.

Changes in dockerfile to launch a script that do the work

This PR requires the PR in development-scripts

### CODE REVIEW

**Comment by trcox** [comment]
@chenchix Any rough performance numbers on this? It's not eliminating a service, so savings would be from # of container reduction, right?

Will have to duplicate change on x86 side Dockerfile prior to merging.

Also, need a way to modify/augment consul CLI args from the compose file via an env, not all deployments will use that argument set (container entrypoints can be customized with compose).

**Comment by chenchix** [comment]
> @chenchix Any rough performance numbers on this? It's not eliminating a service, so savings would be from # of container reduction, right?

More than big performance, it is a matter of avoid to launch and stop core-config-seed everytime as it is a prerequisite in every image in docker-compose, and in a memory limited environment is critical to avoid these peaks.

> Will have to duplicate change on x86 side Dockerfile prior to merging.

Sure, I will do.


> Also, need a way to modify/augment consul CLI args from the compose file via an env, not all deployments will use that argument set (container entrypoints can be customized with compose).

Could you explain this more or suggest a way to do it?

**Comment by trcox** [comment]
@chenchix check out this section on [Docker environment variable replacement](https://docs.docker.com/engine/reference/builder/#environment-replacement). We can set up an ENV CONSUL_ARGS with a default value of `-server -client=0.0.0.0 -bootstrap -ui` and pass the variable to your wrapper script while still allowing users to [set their own arguments at runtime](https://docs.docker.com/engine/reference/run/#env-environment-variables) with the `-e` directive. Additionally, if you start Consul with the included [`docker-entrypoint.sh`](https://github.com/hashicorp/docker-consul/blob/master/0.X/docker-entrypoint.sh) then other settings may also be specified individually.

**Comment by trcox** [comment]
Thanks @chenchix, looking good. Let's move the 2 `*.sh` files under the docker-files directory as well. Once we get the x86 version added I will merge!

**Comment by chenchix** [comment]
@trcox all changes commmited now. Sorry, yesterday I forgot to commit some changes

File: docker-files/Dockerfile [hunk 1]
```diff
@@ -27,17 +27,25 @@ MAINTAINER Cloud Tsai <Cloud.Tsai@Dell.com>
...
 COPY ./config $APP_DIR/config
+COPY docker-files/launch-consul-config.sh $APP_DIR/
 
 #set the working directory
 WORKDIR $APP_DIR
 
-#kick off the tool
-ENTRYPOINT java -jar -Djava.security.egd=file:/dev/urandom -Xmx100M $APP
+#call the wrapper to launch consul and main app
+CMD $APP_DIR/launch-consul-config.shCMD
```
Comment (trcox) [inline docker-files/Dockerfile]: Pop the extra CMD here out and we're all set!

Comment (chenchix) [reply docker-files/Dockerfile]: Fixed!

**Comment by trcox** [comment]
recheck

**Pull request merged**